contacts/api.py

```diff
@@ -0,0 +1,179 @@
+            serializer = UserSerializer(user)
+            return Response(serializer.data, status=HTTP_200_OK)
+        else:
+            return Response("Your token expire Login again", status=HTTP_400_BAD_REQUEST)
+
+    def put(self, request):
+        try:
+            if 'token' in request.COOKIES:
+                token = Token.objects.get(key=request.COOKIES['token'])
+            else:
+                token = None
+        except Token.DoesNotExist:
+            token = None
+        if token is not None:
+            pk = request.COOKIES.get('user_id')
+            user = User.objects.get(id=pk)
+            payload = request.data
+            serializer = UserSerializer(user, data=payload, partial=True)
+            if serializer.is_valid(raise_exception=True):
+                serializer.save()
+            return Response(serializer.data, status=HTTP_200_OK)
+        else:
+            return Response("BAD REQUEST", status=HTTP_400_BAD_REQUEST)
+
+    def get_serialized(self, pk, user):
+
+        queryset = Contact.objects.filter(user_id=pk)
+
+        serializer = ContactSerializer(queryset, many=True)
+        data = serializer.data
+        return data
+
+
+class UserContacts(APIView):
+    permission_classes = (permissions.BasePermission,)
+
+    def get(self, request):
+        try:
+            if 'token' in request.COOKIES:
+                token = Token.objects.get(key=request.COOKIES['token'])
+            else:
+                token = None
+        except Token.DoesNotExist:
+            token = None
+        if token is not None:
+            pk = request.COOKIES.get('user_id')
+            _id = int(pk)
+            user = User.objects.get(id=_id)
+            contact = Contact.objects.filter(user_id=user.id)
+            serializer = ContactSerializer(contact, many=True)
+            return Response(serializer.data)
+        else:
+            return Response("BAD REQUEST")
+
+    def post(self, request):
+        serializer = ContactSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors)
+
+    def put(self, request):
+        try:
+            if 'token' in request.COOKIES:
+                token = Token.objects.get(key=request.COOKIES['token'])
+            else:
+                token = None
+        except Token.DoesNotExist:
+            token = None
+        if token is not None:
+            contact = Contact.objects.get(id=request.data['id'])
+            payload = request.data
+            serializer = ContactSerializer(contact, data=payload, partial=True)
+            if serializer.is_valid(raise_exception=True):
+                serializer.save()
+            return Response(serializer.data, status=HTTP_200_OK)
+        else:
+            return Response("BAD REQUEST", status=HTTP_400_BAD_REQUEST)
```

Comment:
There is a difference between 400 bad request and bad authorization. Please use correct HTTP response codes.

Reply:
```
return Response("UNAUTHORIZED", status=HTTP_401_UNAUTHORIZED)
```